packages/database/src/browser/MigrationManager.js

```diff
@@ -0,0 +1,145 @@
+import { PGlite } from '@electric-sql/pglite';
+import path from 'path';
+import { getConnectionConfig } from './getConnectionConfig';
+
+const DEFAULT_MIGRATIONS_PATH = path.resolve(__dirname, '/packages/database/src/core/migrations');
+
+export class MigrationManager {
+  client = null;
+  config = null;
+
+  constructor(config = {}) {
+    this.config = {
+      ...config,
+      migrationsPath: config.migrationsPath || DEFAULT_MIGRATIONS_PATH,
+      logger: config.logger || {
+        info: message => console.log(message),
+        error: message => console.error(message),
+      },
+    };
+    this.client = new PGlite(getConnectionConfig);
+  }
+
+  async initialize() {
+    await this.createMigrationsTable();
+  }
+
+  async close() {
+    await this.client.close();
+  }
+
+  async createMigrationsTable() {
+    await this.client.exec(`
+      CREATE TABLE IF NOT EXISTS migrations (
+        id SERIAL PRIMARY KEY,
+        version INTEGER NOT NULL,
+        name VARCHAR(255) NOT NULL,
+        executed_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP
+      );
+    `);
+  }
+
+  async loadMigrationFiles() {
+    const migrationFiles = import.meta.glob('../../core/migrations/*.js', { eager: true });
+    const migrations = [];
+
+    const sortedMigrationFiles = Object.fromEntries(
+      Object.entries(migrationFiles).sort((a, b) => a[0].localeCompare(b[0])),
+    );
+
+    for (const [name, migrationModule] of Object.entries(sortedMigrationFiles)) {
+      const targets = migrationModule._meta.targets || [];
+
+      // Only run migration if it is intended for the browser
+      if (!targets.includes('browser')) {
+        continue;
+      }
+
+      const version = migrationModule._meta.version;
+
+      migrations.push({
+        version,
+        name,
+        // Wrap the up/down functions to handle both direct SQL and db.runSql format
+        up: async () => {
+          if (typeof migrationModule.up === 'function') {
+            await migrationModule.up({
+              runSql: sql => this.client.exec(sql),
+            });
+          }
+        },
+        down: async () => {
+          if (typeof migrationModule.down === 'function') {
+            await migrationModule.down({
+              runSql: sql => this.client.exec(sql),
+            });
+          }
+        },
```

Comment:
The existing migrations all use `db.runSql` for querying. So in order to reuse the migrations between server and browser, MigrationManager will act as the db driver, wrap the actual querying method with `runSql`